src/api/status/src/js/github-stats.js

```diff
@@ -1,74 +1,124 @@
-  try {
-    const res = await fetchGitHubApi(owner, repo, 'commits');
-    const commitResponse = await res.json();
-
-    if (!res.ok) throw new Error(`[Error ${res.status}] - ${commitResponse.message}`);
-
-    const lastCommitResponse = commitResponse[0];
-    returnedValue.avatar = lastCommitResponse.author.avatar_url;
-    returnedValue.username = lastCommitResponse.author.login;
-    returnedValue.fullName = lastCommitResponse.commit.author.name;
-    returnedValue.commitURL = lastCommitResponse.html_url;
-    returnedValue.shortSha = lastCommitResponse.sha.substr(0, 7);
-    returnedValue.usernameURL = `https://github.com/${lastCommitResponse.author.login}`;
-  } catch (error) {
-    console.error(error);
-  }
+  const res = await fetchGitHubApi(owner, repo, 'commits');
+  const commitResponse = await res.json();
+  const returnedValue = {};
+
+  if (!res.ok) throw new Error(`[Code ${res.status}] - ${commitResponse.message}`);
+
+  const lastCommitResponse = commitResponse[0];
+  returnedValue.avatar = lastCommitResponse.author.avatar_url || '';
+  returnedValue.username = lastCommitResponse.author.login || '';
+  returnedValue.fullName = lastCommitResponse.commit.author.name;
+  returnedValue.commitURL = lastCommitResponse.html_url;
+  returnedValue.shortSha = lastCommitResponse.sha.substr(0, 7);
+  returnedValue.usernameURL = `https://github.com/${lastCommitResponse.author.login || ''}`;
+
+  await redis.set(`github:commitsInfo-${repo}`, JSON.stringify(returnedValue), 'EX', 10 * 60);
+
+  return returnedValue;
+};
+
+const contributorsInfo = async (owner, repo) => {
+  // Get data from cached data
+  const cached = await redis.get(`github:contributorsInfo-${repo}`);
+  if (cached) return JSON.parse(cached);
 
   // get total contributors: https://docs.github.com/en/rest/reference/repos#list-repository-contributors
-  try {
-    const res = await fetchGitHubApi(owner, repo, 'contributors?per_page=1');
+  const res = await fetchGitHubApi(owner, repo, 'contributors?per_page=1');
+  const returnedValue = {};
 
-    if (res.status >= 400) throw new Error(`[Error ${res.status}] - ${(await res.json()).message}`);
+  if (res.status >= 400) throw new Error(`[Code ${res.status}] - ${(await res.json()).message}`);
 
-    const contributorsResponse = await res.headers.get('link');
-    [, returnedValue.totalContributors] = contributorsResponse.match(
-      /.*"next".*&page=([0-9]*).*"last".*/
-    );
-  } catch (error) {
-    console.error(error);
-  }
+  const contributorsResponse = await res.headers.get('link');
+  [, returnedValue.totalContributors] = contributorsResponse.match(
+    /.*"next".*&page=([0-9]*).*"last".*/
+  );
+
+  await redis.set(`github:contributorsInfo-${repo}`, JSON.stringify(returnedValue), 'EX', 10 * 60);
 
   return returnedValue;
 };
+
+module.exports = async function getGitHubData(owner, repo) {
+  const [statsParticipation_, statsCodeFrequency_, commitsInfo_, contributorsInfo_] =
+    await Promise.all([
```

Comment:
`try/catch` or you'll crash the server when this fails

Reply:
Fixed